src/components/table/daikin-table.ts

```diff
@@ -0,0 +1,408 @@
+  sortable: boolean = false;
+
+  /**
+   * An array of `id`s for the `rows` that have been checked.
+   */
+  @property({ type: Array, attribute: false })
+  selection: string[] = [];
+
+  /**
+   * Sort order of the table.
+   */
+  @property({ type: String, reflect: true })
+  order: "asc" | "desc" | null = null;
+
+  /**
+   * The `key` of the currently sorted column.
+   */
+  @property({ type: String, reflect: true })
+  sort: keyof T | null = null;
+
+  /**
+   * Specify this when you want to customize the sort function.
+   * The function must be ascending, and the return value must be `0`, `1` or `-1`.
+   */
+  @property({ attribute: false })
+  sortFunction:
+    | {
+        [key in keyof T]?: (a: T, b: T, key: key) => number;
+      }
+    | null = null;
+
+  @state()
+  private _bulkRowsCheckState: "unchecked" | "indeterminate" | "checked" =
+    "unchecked";
+
+  /**
+   * The rows displayed in the current table.
+   * Currently this is a sorted `rows`, but pagination may be considered in the future.
+   */
+  @state()
+  private _currentView: T[] = [];
+
+  private _updateCurrentView() {
+    const sort = this.sort;
+    if (!sort) {
+      this._currentView = Array.from(this.rows);
+      return;
+    }
+
+    const sortFunction = this.sortFunction?.[sort] ?? defaultSort;
+    this._currentView = this.rows.toSorted((a, b) => sortFunction(a, b, sort));
+
+    if (this.order === "desc") {
+      this._currentView.reverse();
+    }
+  }
+
+  private _emitChangeCheckEvent() {
+    this.dispatchEvent(new Event("change-check"));
+  }
+
+  private _handleHeaderCheckboxChange(): void {
+    switch (this._bulkRowsCheckState) {
+      case "checked":
+        this.selection = [];
+        break;
+
+      default:
+        this.selection = this._currentView.map(({ id }) => id);
+        break;
+    }
+
+    this._emitChangeCheckEvent();
+  }
+
+  private _handleBodyCheckboxChange(event: Event): void {
+    const { name: id } = event.target as DaikinCheckbox;
+
+    if (this.selection.includes(id)) {
+      this.selection = this.selection.filter((selectedId) => selectedId !== id);
+    } else {
+      this.selection = [...this.selection, id];
+    }
+
+    this._emitChangeCheckEvent();
+  }
+
+  private _handleClickSort(event: Event, key: keyof T): void {
+    event.stopPropagation();
+
+    if (this.sort === key) {
+      this.order = this.order === "asc" ? "desc" : "asc";
+    } else {
+      this.sort = key;
+      this.order = "asc";
+    }
+
+    this.dispatchEvent(new Event("change-sort"));
+  }
+
+  private _updateHeaderCheckboxState() {
+    const selectedIdLengthInPage = this.selection.length;
+
+    this._bulkRowsCheckState =
+      this._currentView.length === selectedIdLengthInPage
+        ? "checked"
+        : selectedIdLengthInPage
+          ? "indeterminate"
+          : "unchecked";
+  }
+
+  override render() {
+    const createHeaderRow = () =>
+      repeat(this.headers, ({ label, alignment, sortable, ...header }) => {
```

Comment:
We have to provide a key to preserve DOM nodes.

```suggestion
      repeat(this.headers, (header) => String(header.key), ({ label, alignment, sortable, ...header }) => {
```